src/visitor.js

```diff
@@ -77,12 +73,12 @@ Visitor.prototype.getId = function() {
 
 Visitor.prototype.getAssignmentRegistry = function() {
   if (!this._assignmentRegistry) {
-    var obj = {};
-    for (var i = 0; i < this._assignments.length; i++) {
-      var assignment = this._assignments[i];
-      obj[assignment.getSplitName()] = assignment;
-    }
-    this._assignmentRegistry = obj;
+    this._assignmentRegistry = this._assignments.reduce((registry, assignment) => {
+      return {
+        ...registry,
+        [assignment.getSplitName()]: assignment
+      };
+    }, {});
```

Comment:
if the last time you wrote a lot of javascript was about 10 years ago, this change was a doozy 😁